### ISSUE TITLE
Fix S4143: False positive when incrementing key using ++ operator

### ISSUE THREAD

**Issue by RoyStevelink** [SonarSource/sonar-csharp#1857]
### Description

When the index key is used and incremented in the same line, Sonar raises issue S4143 on the second usage. However, the used index is not the same for the subsequent call.

### Repro steps
``` csharp
_buffer[_index++] = (byte) (value & 0xFF);
_buffer[_index++] = (byte) ((value >> 8) & 0xFF);     // Issues S4143 raised
```
### Expected behavior

No issue

### Actual behavior

"Major Bug S4143" is raised

### Known workarounds
 - 

### Related information

* SonarC# Version 7.5
* Visual Studio Version 2017 15.8.4
* If running through the Scanner for MSBuild, its version

**Comment by Evangelink** [comment]
Hi @RoyStevelink,

Thanks for the feedback. We managed to reproduce the FP and will work on a fix soon.

Cheers,
Amaury

**Comment by RoyStevelink** [comment]
Thanks!
